_scripts/check.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/sh -e
+if [[ $(which node) && $(node --version) ]]; then
+  node _scripts/check_node_version.js
+else
+  echo "install node to continue installation"
+  echo "https://nodejs.org"
+  exit 1
+fi
+
+if [[ $(which docker) && $(docker --version) ]]; then
+  docker=y
+else
+  docker=n
+fi
+
+os="$(uname -a | cut -f 1 -d ' ')"
+if [ "$os" = "Darwin" ]; then
+    if [ "$docker" = "n" ]; then
+      echo "install docker to continue installation"
+      echo "https://docs.docker.com/docker-for-mac/install/"
+      exit 1
+    fi
+
+    if [[ $(which brew) && $(brew --version) ]]; then
+      brew install gmp
```

Comment:
need to handle the error this can throw:

<img src=https://user-images.githubusercontent.com/128755/42004489-fde9c8d0-7a3d-11e8-993a-c6b299a022ea.png width=300 />